cub/test/catch2_test_triple_chevron.cu

```diff
@@ -0,0 +1,211 @@
+#include <thrust/count.h>
+#include <cub/detail/triple_chevron_launch.cuh>
+#include <cuda/std/tuple>
+#include <cstdio>  // For printf
+
+// Has to go after all cub headers. Otherwise, this test won't catch unused
+// variables in cub kernels.
+#include "catch2_test_cdp_helper.h"
+#include "catch2_test_helper.h"
+
+// %PARAM% TEST_CDP cdp 0:1
+
+template <class T>
+__global__ void mult_two_kernel(const T *d_in, T *d_out, int num_items)
+{
+    const int i = blockIdx.x * blockDim.x + threadIdx.x;
+
+    if(blockIdx.x == 0 && threadIdx.x == 0) {
+        printf("Inside kernel: d_in[%d] = %d, d_out[%d] = %d\n", i, d_in[i], i, d_out[i]);
+    }
+
+    if (i < num_items)
+    {
+        d_out[i] = d_in[i] * T{2};
+        if(blockIdx.x == 0 && threadIdx.x == 0) {
+            printf("After computation: d_out[%d] = %d\n", i, d_out[i]);
+        }
+    }
+}
+
+
+struct cdp_chevron_invoker { 
+  static constexpr int threads_in_block = 256;
+
+  template <class T, class KernelT>
+  CUB_RUNTIME_FUNCTION static cudaError_t invoke(std::uint8_t *d_temp_storage,
+                                                 std::size_t &temp_storage_bytes,
+                                                 KernelT kernel,
+                                                 const T *d_in,
+                                                 T *d_out,
+                                                 int num_items,
+                                                 bool on_device)
+  {
+    NV_IF_TARGET(NV_IS_HOST,
+                 (if (on_device) { return cudaErrorLaunchFailure; }),
+                 (if (!on_device) { return cudaErrorLaunchFailure; }));
+
+    if (d_temp_storage == nullptr)
+    {
+      temp_storage_bytes = static_cast<std::size_t>(num_items);
+      return cudaSuccess;
+    }
+  
+    if (temp_storage_bytes != static_cast<std::size_t>(num_items))
+    {
+      return cudaErrorInvalidValue;
+    }
+
+    const int blocks_in_grid = (num_items + threads_in_block - 1) / threads_in_block;
+
+    return cub::detail::triple_chevron(blocks_in_grid, threads_in_block, 0, 0)
+      .doit(kernel, d_in, d_out, num_items);
+  }
+
+  template <class T>
+  CUB_RUNTIME_FUNCTION static cudaError_t create(std::uint8_t *d_temp_storage,
+                                               std::size_t &temp_storage_bytes,
+                                               const T *d_in,
+                                               T *d_out,
+                                               int num_items,
+                                               bool device_invoke)
+  {
+    return invoke(d_temp_storage,
+                  temp_storage_bytes,
+                  mult_two_kernel<T>,
+                  d_in,
+                  d_out,
+                  num_items,
+                  device_invoke);
+  }
+};
+
+struct cdp_invocable { 
+
+  template <class T>
+  CUB_RUNTIME_FUNCTION cudaError_t operator()(uint8_t* temp, size_t bytes, T d_in, T d_out, int n, bool on_device) const {
```

Comment:
important: your test is failing because chevron invoker modifies local variable `bytes` instead of a reference. This leads to the cdp helper allocating 0 bytes of temporary storage, so `temp` remains `nullptr` and the algorithm thinks it's still at the stage of querying temporary storage size.